### ISSUE TITLE
Environment affects configuration location?

### ISSUE THREAD

**Issue by denisemauldin** [elkeschaper/tral#9]
Hi there!

I installed tral with pip install.  I can use it just fine, but a collaborator using it gets the following error log where it can't seem to find the configuration file and thinks that spec.ini is in the wrong location?  Any thoughts on what might be affecting this?  The github location for the code he's attempting to run is: https://github.com/systemsbiology/dsPyspark

Thanks!

```
Traceback (most recent call last):
  File "/u5/tools/lib/python3.5/site-packages/tral/configuration.py", line 50, in instance
    return self._instance
AttributeError: 'Singleton' object has no attribute '_instance'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/u5/tools/lib/python3.5/site-packages/configobj.py", line 1942, in _handle_configspec
    _inspec=True)
  File "/u5/tools/lib/python3.5/site-packages/configobj.py", line 1229, in __init__
    self._load(infile, configspec)
  File "/u5/tools/lib/python3.5/site-packages/configobj.py", line 1240, in _load
    raise IOError('Config file not found: "%s".' % self.filename)
OSError: Config file not found: "/u5/tools/lib/python3.5/site-packages/tral/data/spec.ini".

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/users/gglusman/proj/dsPyspark/dsSeqModel.py", line 7, in <module>
    from tral.sequence import sequence
  File "/u5/tools/lib/python3.5/site-packages/tral/sequence/sequence.py", line 14, in <module>
    from tral.repeat import repeat, repeat_align
  File "/u5/tools/lib/python3.5/site-packages/tral/repeat/repeat.py", line 17, in <module>
    from tral.repeat import repeat_score, repeat_pvalue, repeat_io
  File "/u5/tools/lib/python3.5/site-packages/tral/repeat/repeat_score.py", line 25, in <module>
    CONFIG_GENERAL = configuration.Configuration.instance().config
  File "/u5/tools/lib/python3.5/site-packages/tral/configuration.py", line 52, in instance
    self._instance = self._decorated()
  File "/u5/tools/lib/python3.5/site-packages/tral/configuration.py", line 74, in __init__
    stringify=True)
  File "/u5/tools/lib/python3.5/site-packages/configobj.py", line 1229, in __init__
    self._load(infile, configspec)
  File "/u5/tools/lib/python3.5/site-packages/configobj.py", line 1325, in _load
    self._handle_configspec(configspec)
  File "/u5/tools/lib/python3.5/site-packages/configobj.py", line 1948, in _handle_configspec
    raise IOError('Reading configspec failed: %s' % e)
OSError: Reading configspec failed: Config file not found: "/u5/tools/lib/python3.5/site-packages/tral/data/spec.ini".
```

**Comment by elkeschaper** [comment]
Dear Denise,

thanks for your question!
During installation, the TRAL config files should be saved in your home folder on your machine. Usually, this would be `~/.tral/`. The missing file `spec.ini` is one of these files. This is described in here:
http://elkeschaper.github.io/tral/install.html

These are the configuration files: https://github.com/elkeschaper/tral/tree/develop/tral/tral_configuration

Perhaps you collegue could make sure she has the `./tral` folder accessible on her machine, with all files included?

Here is a link to the description of the config files:
`http://elkeschaper.github.io/tral/configure.html#configure:` Thanks to you, I just realized the 2nd mentioned configuration file in this online description should really be `spec.ini`, not `confic.ini` for a 2nd time.

Hope this helps! Good luck with everything, and don't hesitate to write if there are still problems!

Elke

**Comment by denisemauldin** [comment]
Hi Elke,

It does help.  Is there a way to force it to create the config files?  When I've installed TRAL via pip install systemwide then people who try to use TRAL end up with this error.  

Thanks,
Denise

**Comment by sbliven** [comment]
This is still a bug. It stems from problems during TRAL installation. If the configuration files don't get copied into your home directory (because you used `python setup.py develop`, or installed as root without setting `--home` (now deprecated for `--tral-home`), or because pip couldn't write to `~/.tral` for some reason), then TRAL throws this unclear error when trying to read the configuration file.

The fix to this should be

- better error messages
- setting `--tral-home` at installation modifies the default config path, allowing alternate config locations
- config files should be downloaded as needed if they aren't available.

**Comment by sbliven** [comment]
I'm now thinking that we will allow the tral data location to be changed at runtime by setting an environmental variable (`TRAL_CONFIG_DIR` or similar).

**Comment by sbliven** [comment]
Will be fixed with #49
